File: string/basic/missing-characters-make-string-pangram/pangram-missing-character.js
const findMissing = s => {
  let alphabets = "abcdefghijklmnopqrstuvwxyz";
  let letterSet = new Set([...alphabets]);
  let string = s.toLowerCase();
  for (let i = 0; i < string.length; i++) {
    letterSet.delete(string[i]);
    if (!letterSet.size) {
      return "nothing missting";
    }
  }
  return [...letterSet].join("");
};

console.log(findMissing("welcome to geeksforgeeks"));

console.log(findMissing("The quick brown fox jumps"));

console.log(findMissing("The quick brown fox jumps over the lazy dog"));

File: string/basic/missing-characters-make-string-pangram/pangram-missing-character.py
def find_missing(s):
    alphabets = "abcdefghijklmnopqrstuvwxyz"
    letter_set = set(list(alphabets))
    s = s.lower()

    for char in s:
        letter_set.discard(char)
        if len(letter_set) == 0:
            return 'nothing missing'

    missting_list = list(letter_set)
    missting_list.sort()
    return ''.join(missting_list)


print(find_missing("welcome to geeksforgeeks"))

print(find_missing("The quick brown fox jumps"))

print(find_missing("The quick brown fox jumps over the lazy dog"))
